velox/exec/Task.cpp

```diff
@@ -2833,6 +2836,66 @@ std::shared_ptr<ExchangeClient> Task::getExchangeClientLocked(
   return exchangeClients_[pipelineId];
 }
 
+std::optional<trace::QueryTraceConfig> Task::maybeMakeTraceConfig() {
+  const auto& queryConfig = queryCtx_->queryConfig();
+  if (!queryConfig.queryTraceEnabled()) {
+    return std::nullopt;
+  }
+
+  traceDirectory_ = queryConfig.queryTraceDir();
+  VELOX_CHECK(
+      !traceDirectory_.empty(),
+      "Query trace enabled but the trace dir is not set.");
+
+  const auto queryTraceNodes = queryConfig.queryTraceNodeIds();
+  if (queryTraceNodes.empty()) {
```

Comment:
I think we just throw here?

Reply:
`queryTraceNodes` would be empty if we only want to record the query metadata.

Reply:
I see. Then we don't need this log? Thanks!